components/VM_Arm/src/main.c

```diff
@@ -395,6 +395,67 @@ static int camkes_vm_utspace_alloc_at(void *data, const cspacepath_t *dest, seL4
 
 }
 
+static bool add_uts(vka_t *vka, seL4_CPtr cap, uintptr_t paddr, size_t size_bits,
+                    bool is_device)
+{
+    allocman_t *allocman = vka->data;
+
+    cspacepath_t path;
+    vka_cspace_make_path(vka, cap, &path);
+
+    /*
+     * The general usage concept for the different UT pools is:
+     *
+     * ALLOCMAN_UT_KERNEL:
+     *   Physical RAM mapped in the kernel windows. This can be used to create
+     *   arbitrary kernel objects.
+     *
+     * ALLOCMAN_UT_DEV_MEM:
+     *   UTs from a device region, which is also usable RAM. This exists because
+     *   on 32-bit platforms the kernel window could be too small to map all
+     *   physical RAM and thus is able to directly access it. Kernel objects can
+     *   be created from these UTs, if the 'canBeDev' parameter in 'alloc' is
+     *   set to true. However, such objects may have restrictions in what it can
+     *   be used for, because the kernel cannot access the content directly.
+     *
+     * ALLOCMAN_UT_DEV:
+     *   Device regions. Such UTs will never be used for an allocation, unless
+     *   explicitly requested by physical address.
+     *
+     * ToDo: The practical distinction between ALLOCMAN_UT_DEV and
+     *       ALLOCMAN_UT_DEV_MEM is, whether the UT can be requested from the
+     *       allocator without providing it's backing physical address. Since
+     *       UTs from the guest RAM region are supposed to be requested using
+     *       the physical address anyway, there seem not reason why they can't
+     *       be in the ALLOCMAN_UT_DEV pool, too.
+     *       The only know use case where this matters is on the NVidia TK1
+     *       platform. It has a SMMU, so in ordere to use DAM in a VM, there is
+     *       no need for VMs to have their RAM addresses match the physical RAM
+     *       addresses (VM config option "map_one_to_one"). However, it is a
+     *       32-bit architecture and the kernel window is too small to cover all
+     *       physical RAM. Thus, the additional RAM is made available via device
+     *       UTs. The allocator needs to be told about this, which is what
+     *       "ram_paddr_base" is used for and and why it's different from
+     *       "ram_base".
+     */
+
+    bool is_guest_ram = false;
+    if (ram_size > 0) {
```

Comment:
We can also use this expression, as it seems overflow safe.
I like the "?:" expression, because you have only one assignment to a variable then. But I expect the compiler to generate the same code from each expression.